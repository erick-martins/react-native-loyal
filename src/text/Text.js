import PropTypes from 'prop-types';
import React from 'react';
import { Text as RNText } from 'react-native';
import ThemeComponent from '../theme/ThemeComponent';

class Text extends ThemeComponent {
  render() {
    const { children, style, ...rest } = this.props;
    const { theme } = this.state;
    const mergedStyles = this.deepMergeStyles(theme, style || {});
    //debugger;
    return (
      <RNText style={mergedStyles} {...rest}>
        {children}
      </RNText>
    );
  }
}

const propTypes = {
  ...RNText.propTypes,
  children: PropTypes.any,
  className: PropTypes.string,
};

Text.propTypes = propTypes;
Text.defaultProps = { className: 'light body' };

/**
 * Headline Convenience
 */
Text.Headline = props => (
  <Text {...ThemeComponent.appendClassName(props, 'headline')}>
    {props.children}
  </Text>
);
Text.Headline.propTypes = propTypes;

/**
 * Title Convenience
 */
Text.Title = props => (
  <Text {...ThemeComponent.appendClassName(props, 'title')}>
    {props.children}
  </Text>
);
Text.Title.propTypes = propTypes;

/**
 * Title 2 Convenience
 */
Text.Title2 = props => (
  <Text {...ThemeComponent.appendClassName(props, 'title2')}>
    {props.children}
  </Text>
);
Text.Title2.propTypes = propTypes;

/**
 * Title 3 Convenience
 */
Text.Title3 = props => (
  <Text {...ThemeComponent.appendClassName(props, 'title3')}>
    {props.children}
  </Text>
);
Text.Title3.propTypes = propTypes;

/**
 * Body Convenience
 */
Text.Body = props => (
  <Text {...ThemeComponent.appendClassName(props, 'body')}>
    {props.children}
  </Text>
);
Text.Body.propTypes = propTypes;

/**
 * Alternative Body Convenience
 */
Text.AltBody = props => (
  <Text {...ThemeComponent.appendClassName(props, 'altBody')}>
    {props.children}
  </Text>
);
Text.AltBody.propTypes = propTypes;

/**
 * Large Body Convenience
 */
Text.LargeBody = props => (
  <Text {...ThemeComponent.appendClassName(props, 'largeBody')}>
    {props.children}
  </Text>
);
Text.LargeBody.propTypes = propTypes;

/**
 * Subtitle Convenience
 */
Text.Subtitle = props => (
  <Text {...ThemeComponent.appendClassName(props, 'subtitle')}>
    {props.children}
  </Text>
);
Text.Subtitle.propTypes = propTypes;

export default Text;
