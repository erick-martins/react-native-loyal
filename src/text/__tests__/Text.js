import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Text from '../Text';

import setupDemoStyle from '../../theme/setupDemoStyle';
setupDemoStyle();

import normalize from '../../helpers/normalizeText';

const DefaultFontFamilies = {
  DEFAULT: 'Open Sans',
  FEATURED: 'Circular',
};
const DefaultFontWeighties = {
  BOOK: '100',
  REGULAR: '300',
  BOLD: '700',
  BLACK: '900',
};

const tests = [
  {
    name:'Headline',
    component:() => <Text className={'headline'} />,
    fontSize:normalize(40),
    fontFamily:DefaultFontFamilies.FEATURED,
    fontWeight:DefaultFontWeighties.BLACK
  },{
    name:'Title',
    component:() => <Text className={'title'} />,
    fontSize: normalize(30),
    fontFamily: DefaultFontFamilies.FEATURED,
    fontWeight: DefaultFontWeighties.BLACK,
  },{
    name:'title2',
    component:() => <Text className={'title2'} />,
    fontSize: normalize(30),
    fontFamily: DefaultFontFamilies.FEATURED,
    fontWeight: DefaultFontWeighties.BOOK,
  },{
    name:'title3',
    component:() => <Text className={'title3'} />,
    fontSize: normalize(20),
    fontFamily: DefaultFontFamilies.FEATURED,
    fontWeight: DefaultFontWeighties.BLACK,
  },{
    name:'body',
    component:() => <Text className={'body'} />,
    fontSize: normalize(13),
    fontFamily: DefaultFontFamilies.DEFAULT,
    fontWeight: DefaultFontWeighties.REGULAR,
  },{
    name:'body bold',
    component:() => <Text className={'body bold'} />,
    fontSize: normalize(13),
    fontFamily: DefaultFontFamilies.DEFAULT,
    fontWeight: DefaultFontWeighties.BLACK,
  },{
    name:'altBody',
    component:() => <Text className={'altBody'} />,
    fontSize: normalize(15),
    fontFamily: DefaultFontFamilies.DEFAULT,
    fontWeight: DefaultFontWeighties.REGULAR,
  },{
    name:'largeBody',
    component:() => <Text className={'largeBody'} />,
    fontSize: normalize(20),
    fontFamily: DefaultFontFamilies.DEFAULT,
    fontWeight: DefaultFontWeighties.REGULAR,
  },{
    name:'subtitle',
    component:() => <Text className={'subtitle'} />,
    fontSize: normalize(10),
    fontFamily: DefaultFontFamilies.DEFAULT,
    fontWeight: DefaultFontWeighties.REGULAR,
  }
];



describe('Text Component', () => {
  it('should render without issues', () => {
    const component = shallow(<Text />);

    //const tree = renderer.create(<Text />).toJSON();
    expect(component).toMatchSnapshot();
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  tests.forEach((testing, idx)=>{
    let testDescription = 'should have font size of '+ testing.fontSize
        +', font weight '+ testing.fontWeight
        +' and font family '+ testing.fontFamily
        +' when '+ testing.name;

    it(testDescription, () => {
      const component = shallow(testing.component());
  
      const style = component.props().style;

      let fontSizeStyle, fontWeightStyle, fontFamilyStyle;
      
      if (style && style.hasOwnProperty('fontSize')) 
          fontSizeStyle = style.fontSize;
      if (style && style.hasOwnProperty('fontWeight')) 
        fontWeightStyle = style.fontWeight;
      if (style && style.hasOwnProperty('fontFamily')) 
        fontFamilyStyle = style.fontFamily;
  
      expect(component).toMatchSnapshot();
      expect(fontSizeStyle).toBe(testing.fontSize);
      expect(fontWeightStyle).toBe(testing.fontWeight);
      expect(fontFamilyStyle).toBe(testing.fontFamily);
    });
  });


});
