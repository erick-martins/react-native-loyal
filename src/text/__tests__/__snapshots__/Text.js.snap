// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component should have font size of 12.5, font weight 300 and font family Open Sans when subtitle 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Text
    accessible={true}
    allowFontScaling={true}
    className="subtitle"
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#63717a",
        "fontFamily": "Open Sans",
        "fontSize": 12.5,
        "fontWeight": "300",
        "letterSpacing": 0.8,
        "lineHeight": 16.25,
      }
    }
  />,
  "nodes": Array [
    <Text
      accessible={true}
      allowFontScaling={true}
      className="subtitle"
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#63717a",
          "fontFamily": "Open Sans",
          "fontSize": 12.5,
          "fontWeight": "300",
          "letterSpacing": 0.8,
          "lineHeight": 16.25,
        }
      }
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Text
      className="subtitle"
    />,
    "_forcedUpdate": false,
    "_instance": Text {
      "context": Object {},
      "props": Object {
        "className": "subtitle",
      },
      "refs": Object {},
      "state": Object {
        "theme": Object {
          "color": "#63717a",
          "fontFamily": "Open Sans",
          "fontSize": 12.5,
          "fontWeight": "300",
          "letterSpacing": 0.8,
          "lineHeight": 16.25,
        },
      },
      "unregister": [Function],
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": Object {
      "theme": Object {
        "color": "#63717a",
        "fontFamily": "Open Sans",
        "fontSize": 12.5,
        "fontWeight": "300",
        "letterSpacing": 0.8,
        "lineHeight": 16.25,
      },
    },
    "_rendered": <Text
      accessible={true}
      allowFontScaling={true}
      className="subtitle"
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#63717a",
          "fontFamily": "Open Sans",
          "fontSize": 12.5,
          "fontWeight": "300",
          "letterSpacing": 0.8,
          "lineHeight": 16.25,
        }
      }
    />,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Text
    className="subtitle"
  />,
}
`;

exports[`Text Component should have font size of 16.25, font weight 300 and font family Open Sans when body 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Text
    accessible={true}
    allowFontScaling={true}
    className="body"
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#63717a",
        "fontFamily": "Open Sans",
        "fontSize": 16.25,
        "fontWeight": "300",
        "letterSpacing": 1.04,
        "lineHeight": 20,
      }
    }
  />,
  "nodes": Array [
    <Text
      accessible={true}
      allowFontScaling={true}
      className="body"
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#63717a",
          "fontFamily": "Open Sans",
          "fontSize": 16.25,
          "fontWeight": "300",
          "letterSpacing": 1.04,
          "lineHeight": 20,
        }
      }
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Text
      className="body"
    />,
    "_forcedUpdate": false,
    "_instance": Text {
      "context": Object {},
      "props": Object {
        "className": "body",
      },
      "refs": Object {},
      "state": Object {
        "theme": Object {
          "color": "#63717a",
          "fontFamily": "Open Sans",
          "fontSize": 16.25,
          "fontWeight": "300",
          "letterSpacing": 1.04,
          "lineHeight": 20,
        },
      },
      "unregister": [Function],
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": Object {
      "theme": Object {
        "color": "#63717a",
        "fontFamily": "Open Sans",
        "fontSize": 16.25,
        "fontWeight": "300",
        "letterSpacing": 1.04,
        "lineHeight": 20,
      },
    },
    "_rendered": <Text
      accessible={true}
      allowFontScaling={true}
      className="body"
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#63717a",
          "fontFamily": "Open Sans",
          "fontSize": 16.25,
          "fontWeight": "300",
          "letterSpacing": 1.04,
          "lineHeight": 20,
        }
      }
    />,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Text
    className="body"
  />,
}
`;

exports[`Text Component should have font size of 16.25, font weight 900 and font family Open Sans when body bold 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Text
    accessible={true}
    allowFontScaling={true}
    className="body bold"
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#63717a",
        "fontFamily": "Open Sans",
        "fontSize": 16.25,
        "fontWeight": "900",
        "letterSpacing": 1.04,
        "lineHeight": 20,
      }
    }
  />,
  "nodes": Array [
    <Text
      accessible={true}
      allowFontScaling={true}
      className="body bold"
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#63717a",
          "fontFamily": "Open Sans",
          "fontSize": 16.25,
          "fontWeight": "900",
          "letterSpacing": 1.04,
          "lineHeight": 20,
        }
      }
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Text
      className="body bold"
    />,
    "_forcedUpdate": false,
    "_instance": Text {
      "context": Object {},
      "props": Object {
        "className": "body bold",
      },
      "refs": Object {},
      "state": Object {
        "theme": Object {
          "color": "#63717a",
          "fontFamily": "Open Sans",
          "fontSize": 16.25,
          "fontWeight": "900",
          "letterSpacing": 1.04,
          "lineHeight": 20,
        },
      },
      "unregister": [Function],
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": Object {
      "theme": Object {
        "color": "#63717a",
        "fontFamily": "Open Sans",
        "fontSize": 16.25,
        "fontWeight": "900",
        "letterSpacing": 1.04,
        "lineHeight": 20,
      },
    },
    "_rendered": <Text
      accessible={true}
      allowFontScaling={true}
      className="body bold"
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#63717a",
          "fontFamily": "Open Sans",
          "fontSize": 16.25,
          "fontWeight": "900",
          "letterSpacing": 1.04,
          "lineHeight": 20,
        }
      }
    />,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Text
    className="body bold"
  />,
}
`;

exports[`Text Component should have font size of 18.75, font weight 300 and font family Open Sans when altBody 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Text
    accessible={true}
    allowFontScaling={true}
    className="altBody"
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#63717a",
        "fontFamily": "Open Sans",
        "fontSize": 18.75,
        "fontWeight": "300",
        "letterSpacing": 1.2,
        "lineHeight": 22.5,
      }
    }
  />,
  "nodes": Array [
    <Text
      accessible={true}
      allowFontScaling={true}
      className="altBody"
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#63717a",
          "fontFamily": "Open Sans",
          "fontSize": 18.75,
          "fontWeight": "300",
          "letterSpacing": 1.2,
          "lineHeight": 22.5,
        }
      }
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Text
      className="altBody"
    />,
    "_forcedUpdate": false,
    "_instance": Text {
      "context": Object {},
      "props": Object {
        "className": "altBody",
      },
      "refs": Object {},
      "state": Object {
        "theme": Object {
          "color": "#63717a",
          "fontFamily": "Open Sans",
          "fontSize": 18.75,
          "fontWeight": "300",
          "letterSpacing": 1.2,
          "lineHeight": 22.5,
        },
      },
      "unregister": [Function],
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": Object {
      "theme": Object {
        "color": "#63717a",
        "fontFamily": "Open Sans",
        "fontSize": 18.75,
        "fontWeight": "300",
        "letterSpacing": 1.2,
        "lineHeight": 22.5,
      },
    },
    "_rendered": <Text
      accessible={true}
      allowFontScaling={true}
      className="altBody"
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#63717a",
          "fontFamily": "Open Sans",
          "fontSize": 18.75,
          "fontWeight": "300",
          "letterSpacing": 1.2,
          "lineHeight": 22.5,
        }
      }
    />,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Text
    className="altBody"
  />,
}
`;

exports[`Text Component should have font size of 25, font weight 300 and font family Open Sans when largeBody 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Text
    accessible={true}
    allowFontScaling={true}
    className="largeBody"
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#63717a",
        "fontFamily": "Open Sans",
        "fontSize": 25,
        "fontWeight": "300",
        "letterSpacing": 1.6,
        "lineHeight": 31.25,
      }
    }
  />,
  "nodes": Array [
    <Text
      accessible={true}
      allowFontScaling={true}
      className="largeBody"
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#63717a",
          "fontFamily": "Open Sans",
          "fontSize": 25,
          "fontWeight": "300",
          "letterSpacing": 1.6,
          "lineHeight": 31.25,
        }
      }
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Text
      className="largeBody"
    />,
    "_forcedUpdate": false,
    "_instance": Text {
      "context": Object {},
      "props": Object {
        "className": "largeBody",
      },
      "refs": Object {},
      "state": Object {
        "theme": Object {
          "color": "#63717a",
          "fontFamily": "Open Sans",
          "fontSize": 25,
          "fontWeight": "300",
          "letterSpacing": 1.6,
          "lineHeight": 31.25,
        },
      },
      "unregister": [Function],
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": Object {
      "theme": Object {
        "color": "#63717a",
        "fontFamily": "Open Sans",
        "fontSize": 25,
        "fontWeight": "300",
        "letterSpacing": 1.6,
        "lineHeight": 31.25,
      },
    },
    "_rendered": <Text
      accessible={true}
      allowFontScaling={true}
      className="largeBody"
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#63717a",
          "fontFamily": "Open Sans",
          "fontSize": 25,
          "fontWeight": "300",
          "letterSpacing": 1.6,
          "lineHeight": 31.25,
        }
      }
    />,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Text
    className="largeBody"
  />,
}
`;

exports[`Text Component should have font size of 25, font weight 900 and font family Circular when title3 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Text
    accessible={true}
    allowFontScaling={true}
    className="title3"
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#63b2db",
        "fontFamily": "Circular",
        "fontSize": 25,
        "fontWeight": "900",
        "lineHeight": 27.5,
      }
    }
  />,
  "nodes": Array [
    <Text
      accessible={true}
      allowFontScaling={true}
      className="title3"
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#63b2db",
          "fontFamily": "Circular",
          "fontSize": 25,
          "fontWeight": "900",
          "lineHeight": 27.5,
        }
      }
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Text
      className="title3"
    />,
    "_forcedUpdate": false,
    "_instance": Text {
      "context": Object {},
      "props": Object {
        "className": "title3",
      },
      "refs": Object {},
      "state": Object {
        "theme": Object {
          "color": "#63b2db",
          "fontFamily": "Circular",
          "fontSize": 25,
          "fontWeight": "900",
          "lineHeight": 27.5,
        },
      },
      "unregister": [Function],
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": Object {
      "theme": Object {
        "color": "#63b2db",
        "fontFamily": "Circular",
        "fontSize": 25,
        "fontWeight": "900",
        "lineHeight": 27.5,
      },
    },
    "_rendered": <Text
      accessible={true}
      allowFontScaling={true}
      className="title3"
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#63b2db",
          "fontFamily": "Circular",
          "fontSize": 25,
          "fontWeight": "900",
          "lineHeight": 27.5,
        }
      }
    />,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Text
    className="title3"
  />,
}
`;

exports[`Text Component should have font size of 37.5, font weight 100 and font family Circular when title2 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Text
    accessible={true}
    allowFontScaling={true}
    className="title2"
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#63b2db",
        "fontFamily": "Circular",
        "fontSize": 37.5,
        "fontWeight": "100",
        "lineHeight": 42.5,
      }
    }
  />,
  "nodes": Array [
    <Text
      accessible={true}
      allowFontScaling={true}
      className="title2"
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#63b2db",
          "fontFamily": "Circular",
          "fontSize": 37.5,
          "fontWeight": "100",
          "lineHeight": 42.5,
        }
      }
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Text
      className="title2"
    />,
    "_forcedUpdate": false,
    "_instance": Text {
      "context": Object {},
      "props": Object {
        "className": "title2",
      },
      "refs": Object {},
      "state": Object {
        "theme": Object {
          "color": "#63b2db",
          "fontFamily": "Circular",
          "fontSize": 37.5,
          "fontWeight": "100",
          "lineHeight": 42.5,
        },
      },
      "unregister": [Function],
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": Object {
      "theme": Object {
        "color": "#63b2db",
        "fontFamily": "Circular",
        "fontSize": 37.5,
        "fontWeight": "100",
        "lineHeight": 42.5,
      },
    },
    "_rendered": <Text
      accessible={true}
      allowFontScaling={true}
      className="title2"
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#63b2db",
          "fontFamily": "Circular",
          "fontSize": 37.5,
          "fontWeight": "100",
          "lineHeight": 42.5,
        }
      }
    />,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Text
    className="title2"
  />,
}
`;

exports[`Text Component should have font size of 37.5, font weight 900 and font family Circular when Title 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Text
    accessible={true}
    allowFontScaling={true}
    className="title"
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#63b2db",
        "fontFamily": "Circular",
        "fontSize": 37.5,
        "fontWeight": "900",
        "lineHeight": 47.5,
      }
    }
  />,
  "nodes": Array [
    <Text
      accessible={true}
      allowFontScaling={true}
      className="title"
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#63b2db",
          "fontFamily": "Circular",
          "fontSize": 37.5,
          "fontWeight": "900",
          "lineHeight": 47.5,
        }
      }
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Text
      className="title"
    />,
    "_forcedUpdate": false,
    "_instance": Text {
      "context": Object {},
      "props": Object {
        "className": "title",
      },
      "refs": Object {},
      "state": Object {
        "theme": Object {
          "color": "#63b2db",
          "fontFamily": "Circular",
          "fontSize": 37.5,
          "fontWeight": "900",
          "lineHeight": 47.5,
        },
      },
      "unregister": [Function],
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": Object {
      "theme": Object {
        "color": "#63b2db",
        "fontFamily": "Circular",
        "fontSize": 37.5,
        "fontWeight": "900",
        "lineHeight": 47.5,
      },
    },
    "_rendered": <Text
      accessible={true}
      allowFontScaling={true}
      className="title"
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#63b2db",
          "fontFamily": "Circular",
          "fontSize": 37.5,
          "fontWeight": "900",
          "lineHeight": 47.5,
        }
      }
    />,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Text
    className="title"
  />,
}
`;

exports[`Text Component should have font size of 50, font weight 900 and font family Circular when Headline 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Text
    accessible={true}
    allowFontScaling={true}
    className="headline"
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#63b2db",
        "fontFamily": "Circular",
        "fontSize": 50,
        "fontWeight": "900",
        "lineHeight": 63.75,
      }
    }
  />,
  "nodes": Array [
    <Text
      accessible={true}
      allowFontScaling={true}
      className="headline"
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#63b2db",
          "fontFamily": "Circular",
          "fontSize": 50,
          "fontWeight": "900",
          "lineHeight": 63.75,
        }
      }
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Text
      className="headline"
    />,
    "_forcedUpdate": false,
    "_instance": Text {
      "context": Object {},
      "props": Object {
        "className": "headline",
      },
      "refs": Object {},
      "state": Object {
        "theme": Object {
          "color": "#63b2db",
          "fontFamily": "Circular",
          "fontSize": 50,
          "fontWeight": "900",
          "lineHeight": 63.75,
        },
      },
      "unregister": [Function],
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": Object {
      "theme": Object {
        "color": "#63b2db",
        "fontFamily": "Circular",
        "fontSize": 50,
        "fontWeight": "900",
        "lineHeight": 63.75,
      },
    },
    "_rendered": <Text
      accessible={true}
      allowFontScaling={true}
      className="headline"
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#63b2db",
          "fontFamily": "Circular",
          "fontSize": 50,
          "fontWeight": "900",
          "lineHeight": 63.75,
        }
      }
    />,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Text
    className="headline"
  />,
}
`;

exports[`Text Component should render without issues 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Text
    accessible={true}
    allowFontScaling={true}
    className="light body"
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#63717a",
        "fontFamily": "Open Sans",
        "fontSize": 16.25,
        "fontWeight": "300",
        "letterSpacing": 1.04,
        "lineHeight": 20,
      }
    }
  />,
  "nodes": Array [
    <Text
      accessible={true}
      allowFontScaling={true}
      className="light body"
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#63717a",
          "fontFamily": "Open Sans",
          "fontSize": 16.25,
          "fontWeight": "300",
          "letterSpacing": 1.04,
          "lineHeight": 20,
        }
      }
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Text
      className="light body"
    />,
    "_forcedUpdate": false,
    "_instance": Text {
      "context": Object {},
      "props": Object {
        "className": "light body",
      },
      "refs": Object {},
      "state": Object {
        "theme": Object {
          "color": "#63717a",
          "fontFamily": "Open Sans",
          "fontSize": 16.25,
          "fontWeight": "300",
          "letterSpacing": 1.04,
          "lineHeight": 20,
        },
      },
      "unregister": [Function],
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": Object {
      "theme": Object {
        "color": "#63717a",
        "fontFamily": "Open Sans",
        "fontSize": 16.25,
        "fontWeight": "300",
        "letterSpacing": 1.04,
        "lineHeight": 20,
      },
    },
    "_rendered": <Text
      accessible={true}
      allowFontScaling={true}
      className="light body"
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#63717a",
          "fontFamily": "Open Sans",
          "fontSize": 16.25,
          "fontWeight": "300",
          "letterSpacing": 1.04,
          "lineHeight": 20,
        }
      }
    />,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Text
    className="light body"
  />,
}
`;

exports[`Text Component should render without issues 2`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  className="light body"
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#63717a",
      "fontFamily": "Open Sans",
      "fontSize": 16.25,
      "fontWeight": "300",
      "letterSpacing": 1.04,
      "lineHeight": 20,
    }
  }
/>
`;
